onnxruntime/core/session/inference_session.h

```diff
@@ -671,6 +677,14 @@ class InferenceSession {
   MemoryProfiler memory_profiler_;
 #endif
 
+#if !defined(ORT_MINIMAL_BUILD) || defined(ORT_MINIMAL_BUILD_CUSTOM_OPS)
+  // Store handles to custom op libraries so that their lifetimes extend the lifetime of the session.
+  //
+  // IMPORTANT: Needs to be declared *before* the session_state_ so that the library handles are released
```

Comment:
>Needs to be declared *before* the session_state_  [](http://example.com/codeflow?start=16&length=49)

I think this is automatically enforced. If one attempts to load a model and the custom lib is not loaded/registered, then the relevant kernels would not be found.